core/src/main/java/nl/nn/adapterframework/util/ClassUtils.java

```diff
@@ -299,7 +301,7 @@ public static String classNameOf(Object o) {
 	 */
 	public static void invokeSetter(Object bean, Method method, String valueToSet) {
 		if(!method.getName().startsWith("set") || method.getParameterTypes().length != 1) {
-			throw new IllegalArgumentException("method must start with [set] and may only contain [1] parameter");
+			throw new IllegalStateException("method must start with [set] and may only contain [1] parameter");
```

Comment:
I'd argue that `IllegalArgumentException` was more appropriate, what was the motivation to change?

Reply:
I agree, however in this case the IllegalArgumentExceptions are meant for the `valueToSet` and not the `method` being invalid.